docs/en/latest/plugin-develop.md

```diff
@@ -437,7 +246,119 @@ function _M.access(conf, ctx)
+
+```
+$(INSTALL) -d $(INST_LUADIR)/apisix/plugins/skywalking
+$(INSTALL) apisix/plugins/skywalking/*.lua $(INST_LUADIR)/apisix/plugins/skywalking/
+```
+
+There are other fields in the `_M` which affect the plugin's behavior.
+
+```lua
+local _M = {
+    ...
+    type = 'auth',
+    run_policy = 'prefer_route',
+}
+```
+
+`run_policy` field can be used to control the behavior of the plugin execution.
+When this field set to `prefer_route`, and the plugin has been configured both
+in the global and at the route level, only the route level one will take effect.
+
+`type` field is required to be set to `auth` if your plugin needs to work with consumer.
+
+## Load plugin and replace plugin
+
+Using `require "apisix.plugins.3rd-party"` will load your plugin, just like `require "apisix.plugins.jwt-auth"` will load the `jwt-auth` plugin.
+
+Sometimes you may want to override a method instead of a whole file. In this case, you can configure `lua_module_hook` in `conf/config.yaml`
+to introduce your hook.
+
+Assumed your configuration is:
+
+```yaml
+apisix:
+    ...
+    extra_lua_path: "/path/to/example/?.lua"
+    lua_module_hook: "my_hook"
+```
+
+The `example/my_hook.lua` will be loaded when APISIX starts, and you can use this hook to replace a method in APISIX.
+The example of [my_hook.lua](https://github.com/apache/apisix/blob/master/example/my_hook.lua) can be found under the `example` directory of this project.
+
+## Check external dependencies
+
+If you have dependencies on external libraries, check the dependent items. if your plugin needs to use shared memory, it
+needs to declare via [customizing Nginx configuration](./customize-nginx-configuration.md), for example :
+
+```yaml
+# put this in config.yaml:
+nginx_config:
+    http_configuration_snippet: |
+        # for openid-connect plugin
+        lua_shared_dict discovery             1m; # cache for discovery metadata documents
+        lua_shared_dict jwks                  1m; # cache for JWKs
+        lua_shared_dict introspection        10m; # cache for JWT verification results
+```
+
+The plugin itself provides the init method. It is convenient for plugins to perform some initialization after
+the plugin is loaded. If you need to clean up the initialization, you can put it in the corresponding destroy method.
+
+Note : if the dependency of some plugin needs to be initialized when Nginx start, you may need to add logic to the initialization
+method "http_init" in the file __apisix/init.lua__, and you may need to add some processing on generated part of Nginx
+configuration file in __apisix/cli/ngx_tpl.lua__ file. But it is easy to have an impact on the overall situation according to the
+existing plugin mechanism, **we do not recommend this unless you have a complete grasp of the code**.
+
+## Encrypted storage fields
+
+::: note
+APISIX version is not less than 3.1
+:::
```

Comment:
<img width="281" alt="image" src="https://github.com/user-attachments/assets/36039c9d-3929-4825-9f40-c795fbf08758" />

I think this sentence can be removed since the doc (next) will be released when 3.13 is out. 3.1. is way back in the days already.

If you think this note is still relevant, there is a formatting issue: it should be `:::note` (without the space); otherwise the admonition will not be rendered properly.